src/worker.jl

```diff
@@ -17,6 +17,25 @@ function worker_init(f::File, options::Dict)
 
             push!(LOAD_PATH, $(project))
 
+            # When a manifest already exists ensure the environment is able to
+            # be precompiled with the version of Julia that is using it. If we
+            # just proceed to import `QuartoNotebookWorker` directly then these
+            # potential errors are not passed back to the server process, thus
+            # hiding the issues.
+            let manifest = Base.project_file_manifest_path(Base.active_project())
+                if isfile(manifest)
+                    pushfirst!(LOAD_PATH, "@stdlib")
+                    import Pkg
+                    popfirst!(LOAD_PATH)
+                    io = IOBuffer()
+                    Pkg.instantiate(; io)
+                    message = String(take!(io))
+                    if contains(message, "ERROR:")
+                        throw(ErrorException(message))
+                    end
+                end
+            end
```

Comment:
This isn't elegant at all. If anyone has ideas on a more robust way to check that the env isn't going to throw errors when loading packages from it that would be great.

Reply:
Thanks, that is simpler.